bin/fba_cmx

```diff
@@ -0,0 +1,817 @@
+            log.error('{:.1f}s\targs.intileid not in {}; exiting'.format(time()-start,fn))
+            sys.exit()
+# AR safe: flavor
+if (args.flavor not in ['dithprec','dithlost','starfaint','science','focus']):
+    log.error('args.flavor not in dithprec,dithlost,starfaint,science,focus; exiting')
+    sys.exit()
+if (args.obscon not in ['dark','bright','dark,bright']):
+    log.error('args.obscon not in dark or bright or "dark,bright"; exiting')
+    sys.exit()
+
+
+# AR is tile in the desi footprint?
+# AR -> if not, special msk and targdir for dithering
+tile_in_desi = is_point_in_desi(dmio.load_tiles(),args.tilera,args.tiledec).astype(int)
+
+if (not tile_in_desi) and (args.flavor=='science'):
+    log.error('requested tile is not in DESI and requested flavor==science; exiting')
+    sys.exit()
+    
+
+# AR safe: outdir
+if (args.outdir[-1]!='/'):
+    args.outdir += '/'
+if (os.path.isdir(args.outdir)==False):
+    os.mkdir(args.outdir)
+    
+root    = '{}{:06d}'.format(args.outdir,args.tileid)
+
+
+# AR dictionary with settings proper to each flavor
+fdict = {}
+# AR minimum number of sky fibres per petal
+fdict['nskypet'] = '40' # AR default
+# AR (no) dithering initialisation
+fdict['seed']    = -1    # AR random seed
+fdict['ndither'] = 0    # AR number of dithered tiles
+fdict['gfrac']   = 0.0    # AR fraction of assigned stars that are dithered with a Gaussian
+fdict['gwidth']  = 0.0    # AR offset [arcsec] per coordinate normal distribution
+fdict['bfrac']   = 0.0    # AR fraction of assigned stars  that are dithered within a box for the dithlost-in-space case
+fdict['bwidth']  = 0.0    # AR box width [arcsec] of the dithering for the dithlost-in-space case
+
+
+    
+# AR flavor settings
+if (args.flavor=='dithprec'):
+    if tile_in_desi==0:
+        fdict['msks']= 'STD_DITHER_GAIA'
+    else:
+        fdict['msks']= 'STD_DITHER'
+    fdict['seed']    = args.seed
+    fdict['ndither'] = 12
+    fdict['gfrac']   = 1.0
+    fdict['gwidth']  = 0.7
+elif (args.flavor=='dithlost'):
+    if tile_in_desi==0:
+        fdict['msks']= 'STD_DITHER_GAIA'
+    else:
+        fdict['msks']= 'STD_DITHER'
+    fdict['seed']    = args.seed
+    fdict['ndither'] = 5
+    fdict['gfrac']   = 0.5
+    fdict['gwidth']  = 2.0
+    fdict['bfrac']   = 0.5
+    fdict['bwidth']  = 10.
+elif (args.flavor=='starfaint'):
+    fdict['msks']   = 'STD_FAINT,SV0_WD'
+    fdict['nskypet']= '100'
+elif (args.flavor=='science'):
+#     if (args.obscon=='dark') : fdict['msks'] = 'SV0_WD,MINI_SV_LRG,MINI_SV_ELG,MINI_SV_QSO'
+#     else:                      fdict['msks'] = 'SV0_WD,MINI_SV_BGS_BRIGHT,SV0_MWS_FAINT'
+    if (args.obscon=='dark') : fdict['msks'] = 'SV0_WD,SV0_LRG,SV0_ELG,SV0_QSO'
+    else:                      fdict['msks'] = 'SV0_WD,SV0_BGS,SV0_MWS_FAINT'
+elif (args.flavor=='focus'):
+    sys.exit('not implemented yet; exiting')
+if (fdict['seed']!=-1):
+    np.random.seed(fdict['seed'])
+
+
+
+# AR directories
+hostname = os.getenv('HOSTNAME')
```

Comment:
This seems fragile- are we only running this script at NERSC and KPNO?  And only on cori?  I guess if this is just a quick-and-dirty thing for the next couple months then that is fine, since the next machine will not be online yet.